fendermint/eth/api/examples/query_blockhash.rs

```diff
@@ -0,0 +1,168 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+//! Example of using the Ethereum JSON-RPC facade with the Ethers provider.
+//!
+//! The example assumes that the following has been started and running in the background:
+//! 1. Fendermint ABCI application
+//! 2. Tendermint Core / Comet BFT
+//! 3. Fendermint Ethereum API facade
+//!
+//! # Usage
+//! ```text
+//! cargo run -p fendermint_eth_api --release --example query_blockhash --
+//! ```
+
+use anyhow::Context;
+use clap::Parser;
+use ethers::{
+    prelude::{abigen, ContractFactory},
+    providers::{Http, JsonRpcClient, Middleware, Provider},
+};
+use ethers_core::{
+    abi::Abi,
+    types::{BlockId, BlockNumber, Bytes, TransactionReceipt, H256, U256, U64},
+};
+use hex;
+use std::{fmt::Debug, path::PathBuf, sync::Arc};
+use tracing::Level;
+
+use crate::common::{adjust_provider, make_middleware, request, TestAccount};
+
+#[allow(dead_code)]
+mod common;
+
+// Generate a statically typed interface for the contract.
+abigen!(QueryBlockhash, "../../testing/contracts/QueryBlockhash.abi");
+
+const QUERYBLOCKHASH_HEX: &'static str =
+    include_str!("../../../testing/contracts/QueryBlockhash.bin");
+
+#[derive(Parser, Debug)]
+pub struct Options {
+    /// The host of the Fendermint Ethereum API endpoint.
+    #[arg(long, default_value = "127.0.0.1", env = "FM_ETH__LISTEN__HOST")]
+    pub http_host: String,
+
+    /// The port of the Fendermint Ethereum API endpoint.
+    #[arg(long, default_value = "8545", env = "FM_ETH__LISTEN__PORT")]
+    pub http_port: u32,
+
+    /// Secret key used to deploy the contract.
+    ///
+    /// Assumed to exist with a non-zero balance.
+    #[arg(long, short)]
+    pub secret_key: PathBuf,
+
+    /// Path to write the contract metadata to.
+    #[arg(long, short)]
+    pub out: Option<PathBuf>,
```

Comment:
You probably don't need this. 

I think with the `flatten` attribute in `clap` we could reuse most of these common options. Not sure what the syntax is now, but it should be possible.